### PR TITLE
[Speculative Decoding]Support different inferseed in speculate decoding

### PR DESCRIPTION
**Pull request by freeliuzc**
## Motivation
1. 目前FD的设计是一整个batch推理时共用一个seed（int），用作sample采样的随机数，每个step自增一次，在非投机解码下，query级别产出的token来自于不同的 seed 
2. 投机解码下，由于每次可能产出两个Drafttoken，相当于每两个token共用一个，这种情况下会降低10%左右的多样性。

## Modifications
1. seed 改为 int* tensor，让batch中每个token的seed都保持不一致

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [x] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/model_executor/layers/sample/sampler.py [hunk 1]
```diff
@@ -56,12 +56,29 @@ def top_p_normalize_probs_paddle(
...
+            seq_len_this_time = seq_lens_this_time[i]
+
+            offsets = 4 * paddle.arange(seq_len_this_time, dtype=topp_seed.dtype)
+            topp_seed[idx : idx + seq_len_this_time, 0] = (
+                topp_seed[idx : idx + seq_len_this_time, 0] + offsets
+            ) % MAX_INFER_SEED
+
+            idx += seq_len_this_time
+        else:
+            idx += 1
```
[the remark this answers is not in the document]
Comment (freeliuzc) [reply fastdeploy/model_executor/layers/sample/sampler.py]: TODO，下个PR优化

File: fastdeploy/model_executor/layers/sample/sampler.py [hunk 2]
```diff
@@ -1064,13 +1082,7 @@ def forward_cuda(
...
         probs = F.softmax(logits)
 
-        top_p, top_k = padding_sampling_params(
-            sampling_metadata.top_p,
-            sampling_metadata.top_k,
-            share_inputs["seq_lens_this_time"],
-            share_inputs["seq_lens_encoder"],
-        )
-        _, next_tokens = top_k_top_p_sampling(probs, top_p=top_p, top_k=top_k, seed=sampling_metadata.seed[0, 0])
+        next_tokens = paddle.argmax(probs, axis=-1)
```
Comment (Deleter-D) [inline fastdeploy/model_executor/layers/sample/sampler.py:1096]: 这里是直接放弃MTP的top_p_top_k_sampling了吗

Comment (freeliuzc) [reply fastdeploy/model_executor/layers/sample/sampler.py:1096]: 目前所有场景下，这个都没问题，先这么提下性能，等过几天，这里会重构下~

**Review by Deleter-D — DISMISSED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]
